luarules/gadgets/unit_shield_behaviour.lua

```diff
@@ -244,36 +247,35 @@ function gadget:GameFrame(frame)
 			shieldUnitIndex[shieldUnitsTotalCount] = shieldUnitID
 		end
 
-		shieldCheckChunkSize = math.max(math.ceil(shieldUnitsTotalCount / 10), 1)
-
-		if lastShieldCheckedIndex > #shieldUnitIndex then
-			lastShieldCheckedIndex = 1
-		end
-
-		shieldCheckEndIndex = math.min(lastShieldCheckedIndex + shieldCheckChunkSize - 1, #shieldUnitIndex)
+		shieldCheckChunkSize = math.max(math.ceil(shieldUnitsTotalCount / 4), 1)
 	end
-
-	for i = lastShieldCheckedIndex, shieldCheckEndIndex do
-		local shieldUnitID = shieldUnitIndex[i]
-		local shieldData = shieldUnitsData[shieldUnitID]
-
-		if shieldData then
-			if not shieldData.shieldCoverageChecked then
-				if shieldData.shieldEnabled then
-					setCoveredUnits(shieldUnitID)
-				else
-					removeCoveredUnits(shieldUnitID)
+	if frame % 11 == 7 then
```

Comment:
What is the reason for offsetting by 7 frames? Is there other processing that runs every 11 frames that you don't want to sync up with?